### PR TITLE
Oops we did it again

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/darwin/secure_transport_tls_channel_handler.c [hunk 1]
```diff
@@ -589,7 +589,7 @@ static int s_increment_read_window(struct aws_channel_handler *handler, struct a
         aws_channel_slot_increment_read_window(slot, window_update_size);
     }
 
-    if (secure_transport_handler->negotiation_finished) {
+    if (secure_transport_handler->negotiation_finished && secure_transport_handler->read_task.node.next) {
```
Comment (JonathanHenson) [inline source/darwin/secure_transport_tls_channel_handler.c]: Are you certain you didn't mean this?

! secure_transport_handler->read_task.node.next

Comment (ColdenCullen) [reply source/darwin/secure_transport_tls_channel_handler.c]: Wow it's amazing this worked

Comment (JonathanHenson) [reply source/darwin/secure_transport_tls_channel_handler.c]: heh... you just lucked out that you never got a partial packet.

**Comment by ColdenCullen** [comment]
[Travis Build](https://travis-ci.org/awslabs/aws-c-io/builds/468226384) actually succeeded, not sure why it's listed as failing.

**Pull request merged**